external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```diff
@@ -137,7 +142,11 @@ public KafkaSpoutConfig(Builder<K, V> builder) {
         /**
          * The kafka spout polls records from the last committed offset, if any. If no offset has been committed it behaves as LATEST
          */
-        UNCOMMITTED_LATEST;
+        UNCOMMITTED_LATEST,
+        /**
+         * Start at the earliest offset whose timestamp is greater than or equal to the given startTimestamp
+         */
+        TIMESTAMP;
```

Comment:
Hmm not sure. I think practically it might make sense to go back a few minutes rather than all the way upto EARLIEST. But that might be a bit out of scope.

Reply:
Not sure I follow. I mean that I think we should have both TIMESTAMP, which unconditionally seeks to a timestamp, and UNCOMMITTED_TIMESTAMP which only seeks to a timestamp if no offset has been committed yet. That way you can use UNCOMMITTED_TIMESTAMP to start at a timestamp the first time you deploy the topology, but the topology doesn't reset when you redeploy.

Reply:
Agreed. Adding that.